### DISCUSSION TITLE
Is there a way to insert a <router-link> instead of a regular link?

### DISCUSSION THREAD

**Discussion by Rusinas** [facebook/lexical#6222] [Q&A]
This is a vue-specific question, I know, but I just started to dig in into this subject and just wanted some help if it was already done somewhere (or anything like this). I know that we can insert whole vue components using `decorators` functionality, but what about inline elements? I haven't tried anything yet, but I think inserting a vue component (which `<router-lnik>` is) as an inline element (a node?) could be problematic

**Comment by KalanaPerera** [comment]
I think it's part of the renderer once you have final html content. 

You will required "flag" to identify if its normal link or  <router-link>  link, then transform data to correct required format.

**Comment by jannikbuschke** [comment]
I don't have a solution but the same problem. My current workaround is to use a normal html anchor element, provide an onclick handler, prevent default behavior by using `event.preventDefault()` `event.stopPropagation()` and invoke my router api programmatically.
